### PR TITLE
tools: add percentiles for the TSO bench latency of the total stats

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Percentiles info is needed in the `pd-tso-bench` tool.

### What is changed and how it works?

Add percentiles for the TSO bench latency of the total stats. By using the tDigest algorithm to reduce the memory and CPU usage.

ref: https://github.com/influxdata/tdigest

### Check List

Tests

- Manual test (add detailed scripts or steps below)

image

```shell
Total:
count: 25546655, max: 47.1519ms, min: 0.1159ms, avg: 2.0932ms
<1ms: 1994494, >1ms: 9096505, >2ms: 14211501, >5ms: 201834, >10ms: 41364, >30ms: 957, >50ms: 0, >100ms: 0, >200ms: 0, >400ms: 0, >800ms: 0, >1s: 0
count: 25546655, <1ms: 7.81%, >1ms: 35.61%, >2ms: 55.63%, >5ms: 0.79%, >10ms: 0.16%, >30ms: 0.00%, >50ms: 0.00%, >100ms: 0.00%, >200ms: 0.00%, >400ms: 0.00%, >800ms: 0.00%, >1s: 0.00%
P0.5: 2.0562ms, P0.8: 2.4417ms, P0.9: 2.7911ms, P0.99: 4.8888ms
```

### Release note

```release-note
None.
```

### CODE REVIEW

File: tools/pd-tso-bench/main.go [hunk 1]
```diff
@@ -218,7 +218,7 @@ func newStats() *stats {
 func (s *stats) update(dur time.Duration) {
 	s.count++
 	s.totalDur += dur
-	s.latencyList = append(s.latencyList, dur.Milliseconds())
+	s.latencyList = append(s.latencyList, float64(dur.Nanoseconds())/1e6)
```
Comment (nolouch) [inline tools/pd-tso-bench/main.go:221]: Will the memory be insufficient?

Comment (JmPotato) [reply tools/pd-tso-bench/main.go:221]: If we have 100000000 values, each of them takes 8 bytes to store, then the total memory usage will be 800MB. I think it's ok if the benchmark doesn't last too long, though the sorting later will be very time-consuming. I think some optimization is needed further.

File: tools/pd-tso-bench/main.go [hunk 2]
```diff
@@ -162,6 +166,9 @@ func showStats(ctx context.Context, durCh chan time.Duration) {
 			fmt.Println("\nTotal:")
 			fmt.Println(total.Counter())
 			fmt.Println(total.Percentage())
+			// Calculate the percentiles by using the tDigest algorithm.
+			fmt.Printf("P0.5: %.4fms, P0.8: %.4fms, P0.9: %.4fms, P0.99: %.4fms", latencyTDigest.Quantile(0.5), latencyTDigest.Quantile(0.8), latencyTDigest.Quantile(0.9), latencyTDigest.Quantile(0.99))
```
Comment (rleungx) [inline tools/pd-tso-bench/main.go:170]: use \n?

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**